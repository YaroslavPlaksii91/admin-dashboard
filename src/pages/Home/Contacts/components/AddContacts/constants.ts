import { IMAGE_SIZE } from '@components/AddPhoto/constants';

export const ADD_CONTACT_FIELDS = Object.freeze({
  PHOTO: 'photo',
  FIRST_NAME: 'firstName',
  LAST_NAME: 'lastName',
  EMAIL: 'email',
  ADDRESS: 'address',
});

export const ADD_CONTACT_CONFIG = Object.freeze({
  [ADD_CONTACT_FIELDS.PHOTO]: {
    label: 'Add photo',
    name: 'photo',
    size: IMAGE_SIZE.BIG,
  },
  [ADD_CONTACT_FIELDS.FIRST_NAME]: {
    label: 'first name',
    type: 'text',
    name: 'firstName',
    placeholder: 'First name',
  },
  [ADD_CONTACT_FIELDS.LAST_NAME]: {
    label: 'last name',
    type: 'text',
    name: 'lastName',
    placeholder: 'Last name',
  },
  [ADD_CONTACT_FIELDS.EMAIL]: {
    label: 'email',
    type: 'email',
    name: 'email',
    placeholder: 'Email',
  },
  [ADD_CONTACT_FIELDS.ADDRESS]: {
    label: 'address',
    type: 'text',
    name: 'address',
    placeholder: 'Address',
  },
});
