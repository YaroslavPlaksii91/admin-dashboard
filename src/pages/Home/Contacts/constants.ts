export const CONTACTS_COLUMNS = Object.freeze([
  {
    title: 'Name',
    name: 'name',
    size: 4,
  },
  {
    title: 'Email',
    name: 'email',
    size: 3,
  },
  {
    title: 'Address',
    name: 'address',
    size: 3,
  },
  {
    title: 'Created at',
    name: 'created',
    size: 2,
  },
]);
