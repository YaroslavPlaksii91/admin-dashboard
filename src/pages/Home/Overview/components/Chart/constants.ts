export const TOOLTIP_TEST_ID = 'tooltip';
