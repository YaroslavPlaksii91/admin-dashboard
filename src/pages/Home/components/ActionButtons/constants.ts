export const SORT_BTN_TEST_ID = 'sortButton';
export const FILTER_BTN_TEST_ID = 'filterButton';
export const ADD_BTN_TEST_ID = 'addButton';
