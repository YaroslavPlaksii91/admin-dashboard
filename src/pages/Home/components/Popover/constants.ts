export const POPOVER_TEST_ID = 'popover';
