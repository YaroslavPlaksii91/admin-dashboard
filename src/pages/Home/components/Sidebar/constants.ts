export const USERS_PAGES = Object.freeze([
  {
    name: 'Overview',
    icon: '/src/assets/icons/sprite.svg#icon-overview',
    link: '/overview',
  },
  {
    name: 'Tickets',
    icon: '/src/assets/icons/sprite.svg#icon-tickets',
    link: '/tickets',
  },
  {
    name: 'Ideas',
    icon: '/src/assets/icons/sprite.svg#icon-ideas',
    link: '/ideas',
  },
  {
    name: 'Contacts',
    icon: '/src/assets/icons/sprite.svg#icon-contacts',
    link: '/contacts',
  },
  {
    name: 'Agents',
    icon: '/src/assets/icons/sprite.svg#icon-agents',
    link: '/agents',
  },
  {
    name: 'Articles',
    icon: '/src/assets/icons/sprite.svg#icon-articles',
    link: '/articles',
  },
]);

export const USERS_SETTINGS = Object.freeze([
  {
    name: 'Settings',
    icon: '/src/assets/icons/sprite.svg#icon-settings',
    link: '/settings',
  },
  {
    name: 'Subscription',
    icon: '/src/assets/icons/sprite.svg#icon-subscription',
    link: '/subscription',
  },
]);

export const SIDEBAR_LINK_TEST_ID = 'sidebarLink';
