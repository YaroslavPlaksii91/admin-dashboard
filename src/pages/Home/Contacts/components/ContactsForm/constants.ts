import { IMAGE_SIZE } from '@components/AddPhoto/constants';

export const CONTACTS_FORM_FIELDS = Object.freeze({
  PHOTO: 'photo',
  FIRST_NAME: 'firstName',
  LAST_NAME: 'lastName',
  EMAIL: 'email',
  ADDRESS: 'address',
});

export const CONTACTS_FORM_CONFIG = Object.freeze({
  [CONTACTS_FORM_FIELDS.PHOTO]: {
    label: 'Add photo',
    name: 'photo',
    size: IMAGE_SIZE.BIG,
  },
  [CONTACTS_FORM_FIELDS.FIRST_NAME]: {
    label: 'first name',
    type: 'text',
    name: 'firstName',
    placeholder: 'First name',
  },
  [CONTACTS_FORM_FIELDS.LAST_NAME]: {
    label: 'last name',
    type: 'text',
    name: 'lastName',
    placeholder: 'Last name',
  },
  [CONTACTS_FORM_FIELDS.EMAIL]: {
    label: 'email',
    type: 'email',
    name: 'email',
    placeholder: 'Email',
  },
  [CONTACTS_FORM_FIELDS.ADDRESS]: {
    label: 'address',
    type: 'text',
    name: 'address',
    placeholder: 'Address',
  },
});
