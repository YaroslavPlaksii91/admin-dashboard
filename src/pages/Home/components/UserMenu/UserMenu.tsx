import { FC, useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
// import { User } from 'firebase/auth';
import { SvgIcon, Typography, Box, Button } from '@mui/material';
import { observer } from 'mobx-react-lite';

import { ROUTES } from '@routes/constants';
// import { endSession } from '@services/localeStorage/localeStorage';
import { initAuthStateListener } from '@services/firebase/firebase';
import { authStore } from '@store/auth';

export const UserMenu: FC = observer(() => {
  const [isLogoutVisible, setIsLogoutVisible] = useState(false);
  // const [currentUser, setCurrentUser] = useState<User | null>(null);
  const navigate = useNavigate();

  useEffect(() => {
    // const unsubscribe = initAuthStateListener(setCurrentUser);
    const unsubscribe = initAuthStateListener(authStore.setUser);

    return () => unsubscribe();
  }, []);

  const handleIconClick = () => {
    setIsLogoutVisible(!isLogoutVisible);
  };

  const handleLogoutClick = () => {
    // endSession();
    authStore.logout();
    navigate(ROUTES.LOGIN_PAGE);
  };

  return (
    <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'end' }}>
      <Button sx={{ mr: '24px' }}>
        <SvgIcon titleAccess="Search" fontSize="small">
          <use href="/src/assets/icons/sprite.svg#icon-search"></use>
        </SvgIcon>
      </Button>
      <Button>
        <SvgIcon titleAccess="New message" fontSize="small">
          <use href="/src/assets/icons/sprite.svg#icon-new"></use>
        </SvgIcon>
      </Button>
      <SvgIcon sx={{ height: '32px', ml: '20px', mr: '20px' }}>
        <use href="/src/assets/icons/sprite.svg#icon-vertical"></use>
      </SvgIcon>
      <Typography
        sx={{
          fontSize: '14px',
          fontWeight: 600,
          lineHeight: 1.43,
          letterSpacing: 0.2,
          mr: '14px',
        }}
      >
        {/* {currentUser && currentUser.displayName} */}
        {authStore.user && authStore.user.displayName}
      </Typography>
      <Button
        sx={{
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          width: '50px',
          height: '50px',
          borderRadius: '50%',
          border: '1px solid #DFE0EB',
        }}
        onClick={handleIconClick}
      >
        <Box
          sx={{
            width: '44px',
            height: '44px',
            borderRadius: '50%',
            overflow: 'hidden',
          }}
        >
          <img
            src="/src/assets/images/users/jones-ferdinand@2x.png"
            alt="User's photo"
          />
        </Box>
      </Button>

      {isLogoutVisible && (
        <Button
          sx={{ position: 'absolute', top: '80px' }}
          onClick={handleLogoutClick}
        >
          Logout
        </Button>
      )}
    </Box>
  );
});
