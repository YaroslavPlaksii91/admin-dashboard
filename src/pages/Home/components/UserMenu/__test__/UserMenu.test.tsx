import { cleanup, render, screen, fireEvent } from '@testing-library/react';
import { User } from 'firebase/auth';

import { ROUTES } from '@routes/constants';
import { authStore } from '@store/auth';

import { UserMenu } from '../UserMenu';

const navigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => navigate,
}));

jest.mock('@store/auth', () => {
  return {
    authStore: {
      isLoggedIn: false,
      user: { displayName: 'John Doe' },
      login: jest.fn(),
      logout: jest.fn(),
      setUser: jest.fn(),
    },
  };
});

jest.mock('@services/firebase/firebase', () => ({
  initAuthStateListener: () => {
    authStore.setUser({ displayName: 'John Doe' } as User);
    return jest.fn();
  },
}));

describe('UserMenu component', () => {
  afterEach(cleanup);

  it('should render correctly', () => {
    const { asFragment } = render(<UserMenu />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render user name and hide logout button by default', () => {
    render(<UserMenu />);

    const userName = screen.getByText('John Doe');
    expect(userName).toBeInTheDocument();

    const logoutButton = screen.queryByText('Logout');
    expect(logoutButton).not.toBeInTheDocument();
  });

  it('should display logout button when user icon is clicked', () => {
    render(<UserMenu />);

    const userIcon = screen.getByAltText("User's photo");
    fireEvent.click(userIcon);

    const logoutButton = screen.getByText('Logout');
    expect(logoutButton).toBeInTheDocument();
  });

  it('should handle logout click', () => {
    render(<UserMenu />);

    const userIcon = screen.getByAltText("User's photo");
    fireEvent.click(userIcon);

    const logoutButton = screen.getByText('Logout');
    expect(logoutButton).toBeInTheDocument();
    fireEvent.click(logoutButton);

    expect(authStore.isLoggedIn).toBe(false);
    expect(navigate).toHaveBeenCalledWith(ROUTES.LOGIN_PAGE);
  });
});
