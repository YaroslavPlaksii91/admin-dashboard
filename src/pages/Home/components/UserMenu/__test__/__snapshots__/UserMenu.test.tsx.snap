// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User menu component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-wfsak8"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1c52rb6-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
      >
        <use
          href="/src/assets/icons/sprite.svg#icon-search"
        />
        <title>
          Search
        </title>
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
      >
        <use
          href="/src/assets/icons/sprite.svg#icon-new"
        />
        <title>
          New message
        </title>
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-42ddvs-MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <use
        href="/src/assets/icons/sprite.svg#icon-vertical"
      />
    </svg>
    <p
      class="MuiTypography-root MuiTypography-body1 css-12plx43-MuiTypography-root"
    >
      John Doe
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14h9gbu-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiBox-root css-16mtpgz"
      >
        <img
          alt="User's photo"
          src="/src/assets/images/users/jones-ferdinand@2x.png"
        />
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
